volestipy/volestipy/src/bindings.cpp

```diff
@@ -119,14 +116,77 @@ double HPolytopeCPP::generate_samples(int walk_len, int number_of_points, int nu
             }
         }
    }
-   
-   std::cout<<"sampling completed"<<std::endl;
+   std::cout<<"Sampling completed"<<std::endl;
 
-// The following block of code should NOT be removed!
+// The following block of code allows us to parse the matrix with the points we are making
    auto n_si=0;
    for (auto it_s = rand_points.begin(); it_s != rand_points.end(); it_s++){
       for (auto i = 0; i != it_s->dimension(); i++){
          samples[n_si++] = (*it_s)[i];
       }
    }
 }
+//////////         end of "generate_samples()"          //////////
+
+
+//////////         start of "rounding()"          //////////
+
+void HPolytopeCPP::rounding(char* rounding_method, double* new_A, double* new_b, double* T_matrix, double* shift, double &round_value){
+
+   // make a copy of the initial HP which will be used for the rounding step
+   auto P(HP);
+   RNGType rng(P.dimension());
+   CheBall = P.ComputeInnerBall();
+   
+   // set the output variable of the rounding step
+   round_result round_res;
+   
+   // walk length will always be equal to 2
+   int walk_len = 2;
+   
+   // run the rounding method 
+   if (strcmp(rounding_method,"min_ellipsoid") == 0){
+      round_res = min_sampling_covering_ellipsoid_rounding<AcceleratedBilliardWalk, MT, VT>(P, CheBall, walk_len, rng);      
+   } else if (strcmp(rounding_method,"svd") == 0){      
+      round_res = svd_rounding<AcceleratedBilliardWalk, MT, VT>(P, CheBall, walk_len, rng);   
+   } else if (strcmp(rounding_method, "max_ellipsoid") == 0){
+      round_res = max_inscribed_ellipsoid_rounding<MT, VT>(P, CheBall); 
+   }
+      
+   // create the new_A matrix
+   MT A_to_copy = P.get_mat();
+   int new_n_hyperplanes = A_to_copy.rows();
+   int new_n_variables = A_to_copy.cols();
```

Comment:
Do you needd `new_n_hyperplanes` and `new_n_variables` since these are the same with HP #hyperplanes and dimension?

Reply:
I thought it would be clearer this way..
If you thing it is more confusing I can change it; as you prefer.

Reply:
I think it is more clear that way

Reply:
Ok I am on it.